sim/validator.py

```diff
@@ -30,25 +30,36 @@ def run(cmd, *args, **kwargs):
             cmd failed with exit code %i
           *****""" % (cmd, returncode))
 
-def simulate_genome(predef_snp_path, reference_path, output_path, num_snps=16000, seed=1):
-    if predef_snp_path:
-        run([
-            "simuG.pl",
-            "-refseq", reference_path,
-            "-snp_vcf", predef_snp_path,
-            # below line tells simuG to also simulate predefined indels.
-            #"-indel_vcf", predef_snp_path, 
-            "-prefix", output_path + "simulated",
-            "-seed", str(seed)
-        ])
-    else:
-        run([
-            "simuG.pl",
-            "-refseq", reference_path,
-            "-snp_count", str(num_snps),
-            "-prefix", output_path + "simulated",
-            "-seed", str(seed)
-        ])
+def simulate_genome_random_snps(cmd, num_snps=16000, seed=1):
+    """ Simulated a genome with random SNPs
+
+        Parameters:
+            cmd (list): List of strings defining the command, see (subprocess.run in python docs)
+            num_snps (int): Number of random SNPs
+            seed (int): Seed value for simulation
+
+        Returns:
+            None
+    """
+    cmd.extend(["-snp_count", str(num_snps),
+                "-seed", str(seed)])
+    run(cmd)
+
+def simulate_genome_from_vcf(cmd, predef_snp_path, seed=1):
```

Comment:
it's inconvenient for users of this code to have to recall the bare-bones command to pass into these functions.

I'd instead write a third function called `simulate_genome` that `simulate_genome_random_snps` and `simulate_genome_from_vcf` call.

Reply:
agreed